tests/components/logger/test_init.py

```diff
@@ -1,143 +1,190 @@
-
-    # Blocked named record child of child
-    assert not log_filter.filter(RECORD("test.child.child", logging.DEBUG))
-
-    # Allowed named record child of child
-    assert log_filter.filter(RECORD("test.child.child", logging.WARNING))
-
-
-async def test_set_filter_empty_config(hass):
-    """Test change log level from empty configuration."""
-    log_filter = await async_setup_logger(hass, NO_LOGS_CONFIG)
+from tests.async_mock import Mock, patch
+
+HASS_NS = "unused.homeassistant"
+COMPONENTS_NS = f"{HASS_NS}.components"
+ZONE_NS = f"{COMPONENTS_NS}.zone"
+GROUP_NS = f"{COMPONENTS_NS}.group"
+CONFIGED_NS = "otherlibx"
+UNCONFIG_NS = "unconfigurednamespace"
+
+
+@pytest.fixture(autouse=True)
+def restore_logging_class():
+    """Restore logging class."""
+    klass = logging.getLoggerClass()
+    yield
+    logging.setLoggerClass(klass)
+
+
+async def test_setting_level(hass):
+    """Test we set log levels."""
+    mocks = defaultdict(Mock)
+
+    with patch("logging.getLogger", mocks.__getitem__):
+        assert await async_setup_component(
+            hass,
+            "logger",
+            {
+                "logger": {
+                    "default": "warning",
+                    "logs": {
+                        "test": "info",
+                        "test.child": "debug",
+                        "test.child.child": "warning",
+                    },
+                }
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert len(mocks) == 4
+
+    assert len(mocks[""].orig_setLevel.mock_calls) == 1
+    assert mocks[""].orig_setLevel.mock_calls[0][1][0] == LOGSEVERITY["WARNING"]
+
+    assert len(mocks["test"].orig_setLevel.mock_calls) == 1
+    assert mocks["test"].orig_setLevel.mock_calls[0][1][0] == LOGSEVERITY["INFO"]
+
+    assert len(mocks["test.child"].orig_setLevel.mock_calls) == 1
+    assert mocks["test.child"].orig_setLevel.mock_calls[0][1][0] == LOGSEVERITY["DEBUG"]
+
+    assert len(mocks["test.child.child"].orig_setLevel.mock_calls) == 1
+    assert (
+        mocks["test.child.child"].orig_setLevel.mock_calls[0][1][0]
+        == LOGSEVERITY["WARNING"]
+    )
 
-    assert logging.getLogger("").isEnabledFor(logging.DEBUG) is False
+    # Test set default level
+    with patch("logging.getLogger", mocks.__getitem__):
+        await hass.services.async_call(
+            "logger", "set_default_level", {"level": "fatal"}, blocking=True
+        )
+    assert len(mocks[""].orig_setLevel.mock_calls) == 2
+    assert mocks[""].orig_setLevel.mock_calls[1][1][0] == LOGSEVERITY["FATAL"]
+
+    # Test update other loggers
+    with patch("logging.getLogger", mocks.__getitem__):
+        await hass.services.async_call(
+            "logger",
+            "set_level",
+            {"test.child": "info", "new_logger": "notset"},
+            blocking=True,
+        )
+    assert len(mocks) == 5
+
+    assert len(mocks["test.child"].orig_setLevel.mock_calls) == 2
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5844602)